### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by EugeneTsalko**

### CODE REVIEW

File: prisma/dev.db
Comment (ak-krivonosov) [inline prisma/dev.db:1]: This file should be added in the .gitignore

File: prisma/schema.prisma [hunk 1]
```diff
@@ -16,6 +16,8 @@ model User {
   email String @unique
   password String
   posts Post[]
+  role String @default("USER")
+  token Token?
```
Comment (ak-krivonosov) [inline prisma/schema.prisma]: The token should not be stored in the DB

File: prisma/schema.prisma [hunk 2]
```diff
@@ -24,4 +26,10 @@ model Post {
   authorId Int
   createdAt DateTime @default(now())
   text String
+}
+
+model Token {
```
Comment (ak-krivonosov) [inline prisma/schema.prisma]: Please delete this model

File: src/controllers/authController.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { type Context } from 'hono';
+import { type UserContext } from '../models/contextModel';
+import { authService } from '../services/authService';
+import { deleteCookie, getCookie, setCookie } from 'hono/cookie';
+
+const cookieMaxAge = 15 * 24 * 60 * 60 * 1000; // 15 days in ms
+
+export const authController = {
+  async signUp(c: UserContext) {
```
Comment (ak-krivonosov) [inline src/controllers/authController.ts]: signUp should simply create a user and return something without tokens. Tokens are necessary for authentication(checking is the user is logged in)

File: src/controllers/authController.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    const userData = c.req.valid('json');
+
+    const newUser = await authService.signUp(userData);
+
+    setCookie(c, 'refreshToken', newUser.refreshToken, {
```
Comment (ak-krivonosov) [inline src/controllers/authController.ts]: Let’s firstly build it without a refresh token

File: src/controllers/authController.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const { email, password } = c.req.valid('json');
+
+    const userData = await authService.signIn({ email, password });
+
+    setCookie(c, 'refreshToken', userData.refreshToken, {
+      maxAge: cookieMaxAge,
+      httpOnly: true,
+    });
+
+    return c.json({ data: { userData }, ok: true }, 201);
```
Comment (ak-krivonosov) [inline src/controllers/authController.ts]: We only need to return a token

File: src/controllers/authController.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  async logOut(c: Context) {
+    const { refreshToken } = getCookie(c);
+    await authService.logOut(refreshToken);
+    deleteCookie(c, 'refreshToken');
+
+    return c.json({ ok: true }, 200);
+  },
+
+  async refresh(c: Context) {
```
Comment (ak-krivonosov) [inline src/controllers/authController.ts]: Let's remove it for now

File: src/dataAcces/authRepository.ts
Comment (ak-krivonosov) [inline src/dataAcces/authRepository.ts:1]: There is nothing from Auth to store in DB. Please delete this file

File: src/middleware/userValidation.ts [hunk 7]
```diff
@@ -23,5 +29,9 @@ export const validateUser = validator('json', (value) => {
     );
   }
 
-  return { id, login, email, password };
+  if (role && !parsedRole.success) {
```
Comment (ak-krivonosov) [inline src/middleware/userValidation.ts:32]: You don’t need this in signup route

File: src/routes/authRoutes.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+import { authController } from '../controllers/authController';
+import { validateSignIn } from '../middleware/authValidation';
+
+export const authRoutes = new Hono();
+
+authRoutes.post('/signup', validateUser, authController.signUp);
+
+authRoutes.post('/signin', validateSignIn, authController.signIn);
+
+authRoutes.post('/logout', authController.logOut);
```
Comment (ak-krivonosov) [inline src/routes/authRoutes.ts]: You should use jwt middleware here to check that you are logged in

File: src/routes/authRoutes.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+authRoutes.get('/refresh', authController.refresh);
```
Comment (ak-krivonosov) [inline src/routes/authRoutes.ts]: Same

File: src/services/authService.ts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+      throw Error(
+        'You need JWT_ACCESS_SECRET and JWT_REFRESH_SECRET environmental variables.'
+      );
+    }
+
+    const accessToken = await sign(
+      { id, email },
+      process.env.JWT_ACCESS_SECRET
+    ); // expired?
+    const refreshToken = await sign(
```
Comment (ak-krivonosov) [inline src/services/authService.ts]: Let's remove it for now

File: src/services/authService.ts [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+      { id, email },
+      process.env.JWT_REFRESH_SECRET
+    );
+
+    return { accessToken, refreshToken };
+  },
+
+  async saveToken(userId: number, refreshToken: string) {
```
Comment (ak-krivonosov) [inline src/services/authService.ts]: Hong/jwt store tokens for you. Please delete this method

File: src/services/authService.ts [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+  async signUp(user: User) {
+    const { login, email, password } = user;
+    const newUser = await userService.create({
+      login,
+      email,
+      password,
+      role: 'USER',
+    });
+
+    const tokens = await this.generateTokens({ id: newUser.id, email });
```
Comment (ak-krivonosov) [inline src/services/authService.ts]: Not sure if we need to log in the user immediately after the registration. Let’s separate these logics for now

**Review by ak-krivonosov — CHANGES_REQUESTED** [review]
Good luck

File: src/routes/authRoutes.ts (hunk 9 above)
Comment (EugeneTsalko) [reply src/routes/authRoutes.ts]: https://github.com/EugeneTsalko/backend-hono/pull/2#discussion_r1428043264
If we remove the refresh method, this route is meaningless and should be removed

**Pull request merged**